src/HotChocolate/Core/src/Types.Scalars/PhoneNumberType.cs

```diff
@@ -0,0 +1,114 @@
+using System.Text.RegularExpressions;
+using HotChocolate.Language;
+
+namespace HotChocolate.Types.Scalars
+{
+    /// <summary>
+    /// A field whose value conforms to the standard E.164 format as specified
+    /// in: https://en.wikipedia.org/wiki/E.164.
+    /// </summary>
+    public class PhoneNumberType : StringType
+    {
+        private static readonly string _validationPattern = @"/^\+[1-9]\d{1,14}$/";
```

Comment:
@PascalSenn @michaelstaib How do you prefer to handle `PhoneNumber` validation? This is the regex i got from https://github.com/Urigo/graphql-scalars/blob/master/src/scalars/PhoneNumber.ts#L3 but maybe it would be better to use https://github.com/twcclegg/libphonenumber-csharp ?